ambry-frontend/src/main/java/com.github.ambry.frontend/AmbryBlobStorageService.java

```diff
@@ -134,36 +141,46 @@ public void shutdown() {
   }
 
   @Override
-  public void handleGet(RestRequest restRequest, RestResponseChannel restResponseChannel) {
+  public void handleGet(final RestRequest restRequest, final RestResponseChannel restResponseChannel) {
     long processingStartTime = System.currentTimeMillis();
     long preProcessingTime = 0;
     handlePrechecks(restRequest, restResponseChannel);
     try {
       logger.trace("Handling GET request - {}", restRequest.getUri());
       checkAvailable();
       RestUtils.SubResource subresource = RestUtils.getBlobSubResource(restRequest);
-      RestRequestMetrics requestMetrics =
-          restRequest.getSSLSession() != null ? frontendMetrics.getBlobSSLMetrics : frontendMetrics.getBlobMetrics;
-      if (subresource != null) {
-        logger.trace("Sub-resource requested: {}", subresource);
-        switch (subresource) {
-          case BlobInfo:
-            requestMetrics = restRequest.getSSLSession() != null ? frontendMetrics.getBlobInfoSSLMetrics
-                : frontendMetrics.getBlobInfoMetrics;
-            break;
-          case UserMetadata:
-            requestMetrics = restRequest.getSSLSession() != null ? frontendMetrics.getUserMetadataSSLMetrics
-                : frontendMetrics.getUserMetadataMetrics;
-            break;
+      String operationOrBlobId =
+          RestUtils.getOperationOrBlobIdFromUri(restRequest, subresource, frontendConfig.frontendPathPrefixesToRemove);
+      if (operationOrBlobId.startsWith("/")) {
+        operationOrBlobId = operationOrBlobId.substring(1);
+      }
+      if (operationOrBlobId.equalsIgnoreCase(OPERATION_GET_PEERS)) {
+        getPeersHandler.handle(restRequest, restResponseChannel,
+            (result, exception) -> submitResponse(restRequest, restResponseChannel, result, exception));
+      } else {
+        RestRequestMetrics requestMetrics =
```

Comment:
Yeah, will look into it - will certainly make the class easier to read and understand.